modules/core/src/v2/coins/stx.ts

```diff
@@ -90,8 +91,25 @@ export class Stx extends BaseCoin {
     return true;
   }
 
+  /**
+   * Check if address is valid, then make sure it matches the base address.
+   *
+   * @param {VerifyAddressOptions} params
+   * @param {String} params.address - the address to verify
+   * @param {String} params.baseAddress - the base address from the wallet
+   */
   isWalletAddress(params: VerifyAddressOptions): boolean {
-    throw new MethodNotImplementedError();
+    const { address, keychains } = params;
+    const pubs = _.map(keychains, (keychain) => accountLib.Stx.Utils.xpubToSTXPubkey(keychain.pub));
+    const addressVersion = accountLib.Stx.Utils.getAddressVersion(address);
+    const baseAddress = accountLib.Stx.Utils.getSTXAddressFromPubKeys(pubs, addressVersion).address;
+    return isSameBaseAddress(this, address, baseAddress);
```

Comment:
Can the code of the `isSameBaseAddress` be pulled inline here? I don't see any other calls to it